src/model/naming.go

```diff
@@ -23,19 +24,96 @@ const (
 
 // Naming of the files.
 type Naming struct {
-	Branch    utils.PathTemplate `json:"branch" validate:"required"`
-	Config    utils.PathTemplate `json:"config" validate:"required"`
-	ConfigRow utils.PathTemplate `json:"configRow" validate:"required"`
+	Branch     utils.PathTemplate `json:"branch" validate:"required"`
+	Config     utils.PathTemplate `json:"config" validate:"required"`
+	ConfigRow  utils.PathTemplate `json:"configRow" validate:"required"`
+	usedLock   *sync.Mutex
```

Comment:
Pridane ... uchovavame si zoznam unikatnych `paths` ... su to 2 `map`, kedze niekedy potrebujeme hladat podla kluca a inokedy podla `path`, ... a takto sa to da bez pouzitia `for` cyklu.